lambda_functions/healthcheck/healthcheck.py

```diff
@@ -0,0 +1,88 @@
+
+import os # isort:skip
+import boto3 # isort:skip
+import logging # isort:skip
+import urllib.request # isort:skip nosec
+
+# Configure logging
+logger = logging.getLogger()
+logger.setLevel(logging.ERROR)
+
+def check_website_health(site, expected_string, env):
+    """
+    check_website_health is a function that calls the site from the event (or environment variables)
+    and checks if the status code is 200 and the expected string exists on that site.
+
+    :param site: Url of site to be checked
+    :type site: string
+    :param expected_string: String of text to be checked
+    :type expected_string: String
+    :param env: This is for us to distinguish the message between the dev, preprod and prod environments
+    :type env: string
+    :raises logger.error: will log an error to the lambda log group
+    :raises logger.error: variable
+    """
+
+    # Ping the site
+    response = urllib.request.urlopen(site) # nosec
+        
+    # Get the status code from the response
+    status_code = response.getcode()
+    response_text = response.read().decode('utf-8')
+
+    # Define metric data
+    cloudwatch = boto3.client('cloudwatch')
+    metric_data = cloudwatch.put_metric_data(
+        MetricData = [
+            {
+                'MetricName': f'{env}-response',
+                'Unit': 'None',
+                'Value': 1
+            },
+        ],
+        Namespace = 'AWS/Lambda'
```

Comment:
I think the namespace wants to be something like SML-Healthcheck.

Can you add: 
Dimensions: [
   Name: 'EndpointUrl'
   Value: '<url>',
   Name: 'ExpectedString',
   Value: '<expected_string>'
]

I am not sure if we want to move the code into the success/error legs - but for nw leave where it is, make these changes and then lets look at the metric in the AWS console together.

Reply:
No problem